### ISSUE TITLE
Implement included_lines and excluded_lines in standard trip creator or OsmConnector

### ISSUE THREAD

**Issue by ialokim** [grote/osm2gtfs#117]
The current PR #99 (Data structure) lacks the implementation of Schedule's [`excluded_lines`](/grote/osm2gtfs/wiki/Schedule#excluded_lines) and [`included_lines`](/grote/osm2gtfs/wiki/Schedule#included_lines).

@AltNico raised the idea of implementing this function already while querying OSM data from the Overpass API to reduce network traffic and memory usage.

**Comment by xamanu** [comment]
In my opinion this should be done in a custom trips creator (current implementation) and not come from a schedule source.

I think this has nothing to do with schedule. If we need it to be configurable I would put it into the config file. But I think it should not be covered by standard behaviour.

**Comment by ialokim** [comment]
> I think this has nothing to do with schedule. If we need it to be configurable I would put it into the config file. But I think it should not be covered by standard behavior.

I'm okay if we move it to the `Configuration`, but why not supporting it out of the box? At least for our Nicaraguan `GTFS`, it would be very useful.

**Comment by xamanu** [comment]
In my opinion it is just not the appropriate place. Why would you do that? Can you give a reasoning?

As far as I see, it is an information that specifies on particular GTFS (for any reason) to omit certain lines coming from the data base. As far as I see this is most typical configuration and hasn't to do with schedule or time information.

**Comment by ialokim** [comment]
> As far as I see this is most typical configuration and hasn't to do with schedule or time information.

Yes, you are right, but why not supporting it within the `config.json` out of the box?

**Comment by xamanu** [comment]
Sure, to put the excluded lines into the config file and handle it in the standard creators seems to me also the right way to go.

Why do you need "included_lines"?

**Comment by ialokim** [comment]
> Why do you need `included_lines`?

Not sure if this is very specific to our Nicaraguan case, but we have a lot of bus services mapped in the `NI` (national) network we don't have any schedule information yet.

As you can see [here](/mapanica/pt-data-nicaragua/blob/237e5c21329fc8c056ef37b5c5319925b8b2c098/timetable.json#L4), instead of _excluding_ all the lines we don't have schedule information (and update this key every time someone added a new line and we want to rerun _osm2gtfs_), we wanted to explicitly _include_ only the lines we want in our GTFS.

**Comment by ialokim** [comment]
Thinking about it another time, I'm not very sure anymore whether this option is really needed: `included_lines` could be deduced from all the lines given in the `schedule.json` and `excluded_lines` shouldn't even be present in `schedule.json`.

The only option I see we could implement to improve things related to this issue, would be __only querying the lines present in `schedule.json` from the OSM servers__.
